Neos.Flow/Classes/Mvc/Controller/Argument.php

```diff
@@ -212,12 +218,15 @@ public function setValue($rawValue): Argument
             $this->value = $rawValue;
             return $this;
         }
+        $configuration = $this->getPropertyMappingConfiguration();
+        $configuredType = $configuration->getConfigurationValue(ObjectConverter::class, ObjectConverter::CONFIGURATION_TARGET_TYPE);
+        if ($configuredType !== null) {
+            $this->dataType = $configuredType;
+        } elseif (isset($rawValue['__type']) && $configuration->getConfigurationValue(ObjectConverter::class, ObjectConverter::CONFIGURATION_OVERRIDE_TARGET_TYPE_ALLOWED) === true) {
+            $this->dataType = $rawValue['__type'];
+        }
         $this->value = $this->propertyMapper->convert($rawValue, $this->dataType, $this->getPropertyMappingConfiguration());
         $this->validationResults = $this->propertyMapper->getMessages();
-        if ($this->validator !== null) {
-            $validationMessages = $this->validator->validate($this->value);
-            $this->validationResults->merge($validationMessages);
-        }
```

Comment:
Maybe we can/should keep those lines. The validator won't be set with the moved `initializeActionMethodValidators()` method in the normal case, so validation will not be executed twice.
However, in case some controller overwrites an argument value for whatever reason, the validation will still be run with that new value, so that's b/c to existing behaviour.